rubix-core/src/main/java/com/qubole/rubix/core/CachingInputStream.java

```diff
@@ -202,28 +203,17 @@ public int read(byte[] buffer, int offset, int length)
     }
     catch (Exception e) {
       log.error(String.format("Failed to read from rubix for file %s position %d length %d. Falling back to remote", remotePath, nextReadPosition, length), e);
-      getParentDataInputStream().seek(nextReadPosition);
-      int read = readFullyDirect(buffer, offset, length);
-      if (read > 0) {
-        nextReadPosition += read;
-        setNextReadBlock();
+      try {
+        getParentDataInputStream().readFully(nextReadPosition, buffer, offset, length);
```

Comment:
Isnt streaming read fallback better for streaming read() call? Ideally we should have the positional read override too here and in it's exception handling readFully is justified.

Reply:
Right, made changes to continue with streaming read fallback here and implemented positioned read API separately.